bundles/nim/init.moon

```diff
@@ -0,0 +1,33 @@
+mode_reg =
+  name: 'nim'
+  extensions: 'nim'
+  create: bundle_load('nim_mode')
+
+howl.mode.register mode_reg
+
+howl.command.register
+  name: 'nim-run'
+  description: 'Compile and run the current Nim file'
+  handler: ->
+    exe = howl.config.nim_executable
+    unless exe
+      error 'Cannot locate nim executable, please define config.nim_executable.'
+    unless howl.io.File(exe).exists
+      error "Nonexistent path '#{exe}' - invalid config.nim_executable."
+    unless howl.app.editor.buffer.file
+      error "No file associated with current buffer."
+    howl.command.run "exec #{exe} c -r " .. howl.app.editor.buffer.file.basename
+
+unload = -> howl.mode.unregister 'nim'
+
+howl.config.define
+  name: 'nim_executable',
```

Comment:
@kirbyfan64 
> But nimsuggest is usually installed somewhere into ~/.nimble...

Oh? The page for nimsuggest you linked to says it doesn't install properly via nimble and suggests compiling it and putting it in the same bin path as nim, which made me believe that was the standard way of doing it.

@shalabhc Even if we keep this variable it could default to just `nim`, right? Provided that `nim` is in the path somewhere already it would still just work then I think.